File: automl_server/app/db/models/tool.py
"""
工具模型
"""
from sqlalchemy import Column, String, Text

from .base_entity import BaseEntity


class ToolModel(BaseEntity):
    """工具模型实体"""
    __tablename__ = "tool_model"

    name = Column(String(255), nullable=False, comment="模型名称")
    model_type = Column(String(50), nullable=True, comment="模型类型")
    endpoint = Column(String(512), nullable=True, comment="服务端点")
    config = Column(Text, nullable=True, comment="配置 JSON")

File: frontend_v2/src/types/tool.ts
/** 工具模型创建请求 */
export interface ToolModelCreate {
  name: string;
  model_type?: string;
  endpoint?: string;
  config?: string;
}

/** 工具模型响应 */
export interface ToolModelResponse {
  id: number;
  name: string;
  model_type: string | null;
  endpoint: string | null;
  config: string | null;
  created_at: string;
}
